internal/k8s/controller.go

```diff
@@ -2277,35 +2278,48 @@ func (lbc *LoadBalancerController) createVirtualServerEx(virtualServer *conf_v1.
 	for _, u := range virtualServer.Spec.Upstreams {
 		endpointsKey := configs.GenerateEndpointsKey(virtualServer.Namespace, u.Service, u.Subselector, u.Port)
 
-		var podEndps []podEndpoint
-		var err error
+		var endps []string
+		if u.UseClusterIP {
```

Comment:
I think it makes sense to refactor the getting endpoints logic into a function so that we can use it both for VS and VSR (prob after this PR) - the code looks identical

Reply:
yeah I thought about that but decided to leave it for later 👍